source/conventions.md

```diff
@@ -171,23 +110,140 @@ lattice expansion where there are [`Fork`](#s:forking) elements and for evaluati
 expressions.
 
 The simplist form of name matching is if the string matches
-the `name` field of an element or elements. Regular expressions can be used. 
-Regular expressions must conform to the [PCRE2](https://www.pcre.org/) standard. 
+the `name` field of an element or elements. 
+For example, the string `"Q1"` will match to all elements named `Q1`.
 
-The names of an element may be "qualified" by prepending a `branch` or `BeamLine` name to the string
-using the string `">>"` as a separator. For example, `"B1>>Qaf.*"` would match
-to all elements in a branch or BeamLine named `"B1"` whose name begins with `"Qaf"`.
+Element matches may be restricted to a given element kind using the notation
+```{code} yaml
+<kind>::<name>
+```
+where `<kind>` is the element kind and `<name>` is the element name.
+Example:
+```{code} yaml
+Marker::bpm.
+```
+This will match to all `Marker` elements whose name is four characters starting with `bpm`.
+
+The `N`{sup}`th` element with a given name can be matched to by appending the character `"#"` 
+followed by an integer `N`. For example, `"Quadrupole::Q1#3"` will match to the third element
+that matches `"Quadrupole::Q1"`. Notice that the `N`{sup}`th` instance selection is applied last. 
```

Comment:
What does "Notice that the `N`-th instance selection is applied last." mean exactly? Could you give an example?

Reply:
Changed in next push (done shortly). I added wording to make this clear.